### PR TITLE
Update Dependabot Scheduled Config

### PR DESCRIPTION
**Pull request by jentfoo**
Update Dependabot with the following:
* Add `jentfoo` to reviewers list
* Remove `crypto` ignore on `api` (no longer using forked version)
* Add configurations for missing gomod paths

Although we are looking to explore Renovate I think we should still correct these gaps soon.  As an alternative I am open to correcting these gaps with Renovate first to help validate Renovate and then just merging this for completeness after.

### CODE REVIEW

**Comment by jentfoo** [comment]
Examples are definitely lower priority.  But if we expect people to reference or start from them, it seems like we should keep them up to date.  That said this PR is expanding the scope fairly dramatically, so I could be convinced to do examples as another iteration.

**Comment by wadells** [comment]
> Do we really need to include everything under `examples`? I'm a bit concerned about how many additional PRs that might produce per week.

Some of the stuff under `examples` has progressed well past an example.  For instance, our virtually all of internal use of teleport depends on `examples/chart`.

**Comment by codingllama** [comment]
+1 for removing examples from automatic updates - I don't want to spend eng time every week touching them up.

> Some of the stuff under `examples` has progressed well past an example. For instance, our virtually all of internal use of teleport depends on `examples/chart`. We could refactor to split out the more `example` stuff from the production grade stuff.

We should certainly do that, calling prod code "examples" is highly misleading. Who is a good person to tag for the refactor?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -38,12 +39,140 @@ updates:
...
+      time: "09:00" # 9am UTC
+    open-pull-requests-limit: 10
+    reviewers:
+      - codingllama
+      - jentfoo
+      - rosstimothy
+      - zmb3
+
+  - package-ecosystem: gomod
+    directory: "/build.assets/tooling/"
```
Comment (codingllama) [inline .github/dependabot.yml:76]: nit:

```suggestion
    directory: "/build.assets/tooling"
```

**Review by codingllama — COMMENTED** [review]
Overall looks good, thanks Jent.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -38,12 +39,62 @@ updates:
       interval: weekly
       day: "sunday"
       time: "09:00" # 9am UTC
-    ignore:
-      # Forked/replaced dependencies
-      - dependency-name: golang.org/x/crypto
```
Comment (codingllama) [inline .github/dependabot.yml:43]: Nice catch.

**Review by codingllama — APPROVED** [review]

**Comment by jentfoo** [comment]
I have removed `examples` from this PR.  II still do believe we need a continual update process for `examples`, but I don't want that to block the rest of this and I have no concerns figuring it out in another iteration.

I will try to today to get a few more PR's up to update these dependencies before the scheduled run (reduce the Dependabot qty)

**Review by rosstimothy — APPROVED** [review]
Thanks @jentfoo!

**Comment by codingllama** [comment]
> I still do believe we need a continual update process for examples

I kind of disagree here, I don't think examples should create a continuous maintenance burden. It's important that they remain useful (as in they work with current Teleport), but a pre-release check could do. Touching up their dependencies, barring for CVEs, doesn't seem important.

**Pull request merged**